proof/crefine/AARCH64/IpcCancel_C.thy

```diff
@@ -1336,370 +1174,382 @@ lemma cbitmap_L2_relation_bit_clear:
-           apply (frule_tac s=\<sigma> in tcb_queue_relation_prev_next')
-               apply (fastforce simp add: ksQ_tcb_at')+
-           apply (drule_tac s=\<sigma> in tcbSchedDequeue_update, assumption,
-                  simp_all add: remove1_filter ksQ_tcb_at')[1]
-           apply (clarsimp simp: filter_noteq_op upd_unless_null_def)
-           apply (rule conjI, solves \<open>clarsimp simp: le_maxDomain_eq_less_numDomains
-                                                     unat_trans_ucast_helper\<close>)
-           apply (clarsimp simp: maxDomain_le_unat_ucast_explicit)
-           apply (rule conjI, clarsimp)
-            apply (clarsimp simp: h_val_field_clift'
-                                  h_t_valid_clift[THEN h_t_valid_field] h_t_valid_clift)
-            apply (clarsimp simp: typ_heap_simps)
-            apply (rule conjI; clarsimp simp: typ_heap_simps)
-             apply (erule_tac S="set (ksReadyQueues \<sigma> (tcbDomain ko, tcbPriority ko))"
-                      in state_relation_queue_update_helper',
-                    (simp | rule globals.equality)+,
-                    simp_all add: clift_field_update if_Some_helper numPriorities_def
-                                  cready_queues_index_to_C_def2
-                                  maxDom_to_H maxPrio_to_H
-                            cong: if_cong split del: if_split)[1]
-                apply (fastforce simp: tcb_null_sched_ptrs_def)
-               apply (fastforce simp: typ_heap_simps c_guard_clift)
-              apply (fastforce simp: typ_heap_simps)
-             apply (fastforce simp: tcb_null_sched_ptrs_def)
-            apply (erule_tac S="set (ksReadyQueues \<sigma> (tcbDomain ko, tcbPriority ko))"
-                     in state_relation_queue_update_helper',
-                   (simp | rule globals.equality)+,
-                   simp_all add: clift_field_update if_Some_helper numPriorities_def
-                                 cready_queues_index_to_C_def2
-                                 maxDom_to_H maxPrio_to_H
-                           cong: if_cong split del: if_split)[1]
-               apply (fold_subgoals (prefix))[4]
-            subgoal premises prems using prems
-              by - (fastforce simp: typ_heap_simps c_guard_clift tcb_null_sched_ptrs_def
-                                    clift_heap_update_same[OF h_t_valid_clift])+
-           apply (clarsimp)
-           apply (rule conjI; clarsimp simp: typ_heap_simps)
-            apply (rule conjI)
-             apply (fastforce simp: c_invert_assist l2BitmapSize_def' wordRadix_def)
-            apply (rule conjI; clarsimp)
-             (* invalid, missing bitmap updates on haskell side *)
-             apply (drule tcb_queue_relation'_empty_ksReadyQueues)
-              apply (fold_subgoals (prefix))[2]
-             subgoal premises prems using prems by (fastforce elim: obj_at'_weaken)+
-            (* invalid, missing bitmap updates on haskell side *)
```

Comment:
same comment as above on disappearing comments; maybe I'm missing something, did this get factored out?